libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -151,6 +151,40 @@ void ArduinoOTAClass::begin(bool useMDNS) {
 #endif
 }
 
+/**
+ * Generates nonce using PRNG / TRNG if available
+ * @return  Random MD5 string (nonce)
+ */
+String ArduinoOTAClass::generateNonce(void) {
+
+  MD5Builder nonce_md5;
+  nonce_md5.begin();
+
+  // Use ESP8266 PRNG via RANDOM_REG32 register
+  #if defined(ESP8266)
```

Comment:
AFAIK, this code isn't shared with ESP32. I just checked, and their code base is currently very different.
Therefore, it doesn't make sense to have a specific case here for ESP32.
The same code change can be proposed to them in their repo.

In addition, I understand that this code is also specific to the ESP8266. Therefore, it doesn't make sense to consider a case where the ESP8266 macro is not defined.

I suggest simplifying the proposal to the specific case of ESP8266.

A side note: I also suggest implementing:
`inline uint32_t esp_random() {return RANDOM_REG32;}`
somewhere.

Reply:
@devyte Got it, I'll keep ESP8266 option only. Thank you!

Just found out that ESP8266 Arduino Core has a `secureRandom()` function which is defined in `Arduino.h` and implemented in `WMath.cpp`, so there is a conflict between with the ESP32 `esp_random()`. Maybe it'll be more conventional to implement the `secureRandom()` function in the ESP32, instead of adding `esp_random()` to ESP8266?